test/hotspot/jtreg/vmTestbase/gc/gctests/StringInternGC/StringInternGC.java

```diff
@@ -46,13 +46,25 @@
  * String pool should not overflow.
  */
 public class StringInternGC extends ThreadedGCTest {
-        private int maxLength = 1000;
+        private int maxLength = 1000; // Maximum number of characters to add per operation.
+        private int maxTotalLength = 128 * 1024; // Total maximum length of the string until a new StringBuffer will be allocated.
+        private long lastTime = System.currentTimeMillis();
 
         private class StringGenerator implements Runnable {
                 private StringBuffer sb = new StringBuffer();
 
                 private String generateString() {
                         int length = LocalRandom.nextInt(maxLength);
+                        if (sb.length() > maxTotalLength) {
+                                sb = new StringBuffer();
+                        }
+
+                        long currentTime = System.currentTimeMillis();
```

Comment:
Please move the `system.gc()` call (and the time calculation) into the caller of this method.  Having a method called `generateString()` do a `system.gc()` seems quite surprising.

Reply:
> Please move the `system.gc()` call (and the time calculation) into the caller of this method. Having a method called `generateString()` do a `system.gc()` seems quite surprising.

Updated.
Thanks.